test/conformance/defaultbackend/feature.go

```diff
@@ -66,38 +68,74 @@ func theIngressStatusShowsTheIPAddressOrFQDNWhereItIsExposed() error {
 	return godog.ErrPending
 }
 
-func iSendARequestToHttp(arg1 string, arg2 string, arg3 string) error {
-	return godog.ErrPending
+func iSendARequestToHttp(method string, hostname string, path string) error {
+	return state.CaptureRoundTrip(method, "http", hostname, path)
 }
 
-func theResponseStatuscodeMustBe(arg1 int) error {
-	return godog.ErrPending
+func theResponseStatuscodeMustBe(statusCode int) error {
+	return state.AssertStatusCode(statusCode)
 }
 
-func theResponseMustBeServedByTheService(arg1 string) error {
-	return godog.ErrPending
+func theResponseMustBeServedByTheService(service string) error {
+	return state.AssertServedBy(service)
 }
 
-func theResponseProtoMustBe(arg1 string) error {
-	return godog.ErrPending
+func theResponseProtoMustBe(proto string) error {
+	return state.AssertResponseProto(proto)
 }
 
-func theResponseHeadersMustContainKeyWithMatchingValue(arg1 *messages.PickleStepArgument_PickleTable) error {
-	return godog.ErrPending
+func theResponseHeadersMustContainKeyWithMatchingValue(headers *messages.PickleStepArgument_PickleTable) error {
+	if len(headers.Rows) < 1 {
+		return fmt.Errorf("expected a table with at least one row")
+	}
+
+	for i, row := range headers.Rows {
+		if i == 0 {
+			// Skip the header row
```

Comment:
Is there a way to make this more obvious, e.g. row must match a particular format, otherwise error.
Or omit this behavior and just live without the header...